google/cloud/spanner/internal/time.cc

```diff
@@ -0,0 +1,216 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/internal/time.h"
+#include <chrono>
+#include <cstdint>
+#include <cstring>
+#include <ctime>
+#include <iomanip>
+#include <ratio>
+#include <sstream>
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+
+// All the civil-time code assumes the proleptic Gregorian calendar, and
+// 24-hour days divided into 60-minute hours and 60-second minutes.
+
+using nanoseconds = std::chrono::nanoseconds;
+using seconds = std::chrono::seconds;
+using system_clock = std::chrono::system_clock;
+using time_point = system_clock::time_point;
+
+//
+// Duration
+//
+
+google::protobuf::Duration ToProto(nanoseconds ns) {
+  auto nanos = ns.count();
+  google::protobuf::Duration proto;
+  proto.set_seconds(nanos / nanoseconds::period::den);  // rounds toward zero
+  proto.set_nanos(nanos % nanoseconds::period::den);    // (a/b)*b + a%b == a
+  return proto;
+}
+
+void ToProto(nanoseconds ns, google::protobuf::Duration* proto) {
+  *proto = ToProto(ns);
```

Comment:
In case you care, this function is not tested.

Reply:
Removed.